app/controllers/dogs_controller.rb

```diff
@@ -1,2 +1,27 @@
 class DogsController < ApplicationController
+  def index
+    # @dogs = Dog.by_breed_and_personality(params[:breed], params[:personality])
+    @dogs = Dog.all
+  end
+
+  def new
+    @dog = Dog.new
+  end
+
+  def create
+    @dog = Dog.new(dogs_params)
+    # TODO: this is just a placeholder, please replace with current user later on.
+    @dog.owner = User.first
+    if @dog.save
+      # Redirect to root_path for now, change to show path once that is done.
+      redirect_to root_path
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  private
```

Comment:
Where on Earth is the space after Private!!!!!!  😱
```suggestion
  private
  
```